stencil-library/src/components/business-details/business-details.tsx

```diff
@@ -0,0 +1,95 @@
+import { Component, Host, Prop, State, h } from '@stencil/core';
+import { IBusiness } from '../../api/Business';
+import { Api } from '../../api';
+import { ErrorState, LoadingState } from '../details/utils'; // Make sure to adjust the path if necessary
+
+enum RENDER_STATES {
+  LOADING = 'loading',
+  READY = 'ready',
+  ERROR = 'error',
+}
+
+/**
+ *
+ * @exportedPart detail-section
+ * @exportedPart detail-section-title
+ * @exportedPart detail-section-item-title
+ * @exportedPart detail-section-item-data
+ * @exportedPart detail-empty-state
+ */
+@Component({
+  tag: 'justifi-business-details',
+  styleUrl: 'business-details.scss',
+  shadow: true,
+})
+export class BusinessDetails {
+  @Prop() businessId: string;
+  @Prop() authToken: string;
+  @State() business: IBusiness;
+  @State() renderState: RENDER_STATES = RENDER_STATES.LOADING;
+  @State() errorMessage: string = 'An error ocurred.';
+
+  private api: any;
+
+  constructor() {
+    this.fetchBusiness = this.fetchBusiness.bind(this);
+    this.setError = this.setError.bind(this);
+  }
+
+  async componentWillLoad() {
+    if (!this.authToken) {
+      this.setError(
+        'Missing auth-token. The form will not be functional without it.',
+      );
+      return;
+    }
+
+    if (!this.businessId) {
+      this.setError(
+        'Missing business-id. The form will not be functional without it.',
+      );
+      return;
+    }
+
+    this.api = Api(this.authToken, process.env.ENTITIES_ENDPOINT);
+    await this.fetchBusiness(this.businessId);
+  }
+
+  async fetchBusiness(businessId) {
+    this.renderState = RENDER_STATES.LOADING;
+    try {
+      const response = await this.api.get(`entities/business/${businessId}`);
+      if (response.error) {
+        this.setError(response.error.message);
+        return;
+      }
+      this.business = response.data;
+      this.renderState = RENDER_STATES.READY;
+    } catch (error) {
+      this.setError(error);
+    }
+  }
+
+  setError(error: string) {
+    this.errorMessage = error || this.errorMessage;
+    this.renderState = RENDER_STATES.ERROR;
+  }
+
+  render() {
+    if (this.renderState === RENDER_STATES.LOADING) {
+      return LoadingState;
+    }
+
+    if (this.renderState === RENDER_STATES.ERROR) {
+      return ErrorState(this.errorMessage);
+    }
+
+    return (
+      <Host>
+        <business-details-section business={this.business} />
```

Comment:
Can we be consistent with our business form sections? The mocks may only be 3 sections but I think we should break it up in the same way that we did our business form:
```
business-generic-info
legal-address-form
business-representative
business-owners
additional-questions
```
which would be:
```
business-generic-info-details
legal-address-form-details
business-representative-details
business-owners-details
additional-questions-details
```

@nhalm are you okay with that? I think it strays from the mocks but it gives us more modularity and the possibility to "hot swap" sections in the case we would want to offer a UI with read only and read/write sections, or if a customer wanted to create their own. Generally probably good to keep them aligned.

Reply:
I think that makes perfect sense. I'll make the changes.